arbitrator/prover/test-cases/bulk-memory.wat

```diff
@@ -0,0 +1,54 @@
+;; Copyright 2023, Offchain Labs, Inc.
+;; For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE
+
+(module
+    (import "env" "wavm_halt_and_set_finished" (func $halt))
+    (func $start
+        ;; test memory_fill
+        (memory.fill (i32.const 0x1003) (i32.const 5) (i32.const 4)) ;; ---5555---
+        (memory.fill (i32.const 0x1001) (i32.const 8) (i32.const 3)) ;; -888555---
+        (memory.fill (i32.const 0x1005) (i32.const 2) (i32.const 1)) ;; -888525---
+        (memory.fill (i32.const 0x1001) (i32.const 9) (i32.const 0)) ;; -888525---
+        (memory.fill (i32.const 0x1000) (i32.const 0) (i32.const 0)) ;; -888525---
+        (call $check (i32.const 0x1000) (i32.const 0))
+        (call $check (i32.const 0x1001) (i32.const 8))
+        (call $check (i32.const 0x1002) (i32.const 8))
+        (call $check (i32.const 0x1003) (i32.const 8))
+        (call $check (i32.const 0x1004) (i32.const 5))
+        (call $check (i32.const 0x1005) (i32.const 2))
+        (call $check (i32.const 0x1006) (i32.const 5))
+        (call $check (i32.const 0x1007) (i32.const 0))
+
+        ;; test memory_copy
+        (memory.copy (i32.const 0x1008) (i32.const 0x1000) (i32.const 8))  ;; -888525--888525-----------------
+        (memory.copy (i32.const 0x1009) (i32.const 0x1004) (i32.const 4))  ;; -888525--525-25-----------------
+        (memory.copy (i32.const 0x1009) (i32.const 0x1009) (i32.const 0))  ;; -888525--525-25-----------------
+        (memory.copy (i32.const 0x1009) (i32.const 0x1009) (i32.const 1))  ;; -888525--525-25-----------------
+        (memory.copy (i32.const 0x1009) (i32.const 0x100a) (i32.const 1))  ;; -888525--225-25-----------------
+        (memory.copy (i32.const 0x100f) (i32.const 0x1001) (i32.const 1))  ;; -888525--225-258----------------
+        (memory.copy (i32.const 0x100f) (i32.const 0x1000) (i32.const 32)) ;; ----------------888525--225-258-
+        (memory.copy (i32.const 0x1001) (i32.const 0x100f) (i32.const 32)) ;; --888525--225-258---------------
```

Comment:
I think this does properly test both forward and backwards overlapping copies, but to make sure, could you replace the first check in memory_copy to determine if the copy should go backwards or forwards with an `i32.const 0` and `i32.const 1` and make sure this test fails in either case? Thanks